Source/Core/VideoCommon/TextureCacheBase.cpp

```diff
@@ -636,8 +636,11 @@ class ArbitraryMipmapDetector
         ptr1 += shape.row_length;
         ptr2 += shape.row_length;
       }
+      // calculate the MSE over all pixels, divide by 2.56 to make it a percent
+      // (IE scale to 0..100 instead of 0..256)
 
-      return (float)current_diff_sum / (shape.width * shape.height * 4) / 2.56f;
+      return std::sqrt(static_cast<float>(current_diff_sum)) / (shape.width * shape.height * 4) /
```

Comment:
The sqrt should be around all of `static_cast<float>(current_diff_sum)) / (shape.width * shape.height * 4)`, not just `static_cast<float>(current_diff_sum)`, otherwise the values get scaled really tiny and don't make sense.